Dockerfile

```diff
@@ -1,23 +1,26 @@
-FROM node:10-jessie
+FROM --platform=${TARGETPLATFORM:-linux/amd64} node:15.2.1-alpine3.12
 
-RUN apt-get update -y && \
-    apt-get install -y gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget libgbm1 && \
-    apt-get clean && rm -rf /tmp/* /var/lib/apt/lists/* /var/tmp/*
-RUN mkdir -p /home/node/cloudproxy && chown -R node:node /home/node/cloudproxy
-WORKDIR /home/node/cloudproxy
+# Print build information
+ARG TARGETPLATFORM
+ARG BUILDPLATFORM
+RUN printf "I am running on ${BUILDPLATFORM:-linux/amd64}, building for ${TARGETPLATFORM:-linux/amd64}\n$(uname -a)\n"
 
-COPY package*.json ./
-USER node
-RUN PUPPETEER_PRODUCT=chrome npm install
-COPY --chown=node:node . .
+# Install packages
+RUN apk add --no-cache chromium
 
-ENV LOG_LEVEL=info
-ENV LOG_HTML=
-ENV PORT=8191
-ENV HOST=0.0.0.0
+# Copy CloudProxy code
+USER node
+RUN mkdir -p /home/node/cloudproxy
+WORKDIR /home/node/cloudproxy
+COPY --chown=node:node package.json ./
+COPY --chown=node:node src ./src/
 
-# ENV CAPTCHA_SOLVER=harvester|<more coming soon>...
-# ENV HARVESTER_ENDPOINT=https://127.0.0.1:5000/token
+# Install package. Skip installing Chrome, we will use the installed package.
+ENV PUPPETEER_PRODUCT=chrome \
+    PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true \
+    PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser
+RUN npm install --only=production && \
+    npm install hcaptcha-solver --only=production

```

Comment:
I'm no node expert, but shouldn't adding `hcaptcha-solver` to `packages.json` be sufficient to automatically installing with `npm install` ?

Reply:
I think the original idea was not to install it unless the used needed it. But I'm betting that most people will want it seeing as how CF has now switched over to hCaptch almost exclusively.

Reply:
Yes, but if we include `hcaptcha-solver` in the package.json it's downloaded for all users. Now is optional if you install with npm and included in the Docker.
I think it only takes 1 or 2 MB disk space so we can make it mandatory in other PR if you want.